vendor/github.com/shirou/gopsutil/internal/common/common_linux.go

```diff
@@ -9,15 +9,24 @@ import (
 )
 
 func DoSysctrl(mib string) ([]string, error) {
-	err := os.Setenv("LC_ALL", "C")
-	if err != nil {
-		return []string{}, err
+	hostEnv := os.Environ()
+	foundLC := false
+	for i, line := range hostEnv {
+		if strings.HasPrefix(line, "LC_ALL") {
+			hostEnv[i] = "LC_ALL=C"
+			foundLC = true
+		}
+	}
+	if !foundLC {
+		hostEnv = append(hostEnv, "LC_ALL=C")
 	}
 	sysctl, err := exec.LookPath("/sbin/sysctl")
 	if err != nil {
 		return []string{}, err
 	}
-	out, err := exec.Command(sysctl, "-n", mib).Output()
+	cmd := exec.Command(sysctl, "-n", mib)
+	cmd.Env = hostEnv
```

Comment:
I'm not sure I understand what you're trying to say. This change sets LC_ALL=C just for this subprocess that needs it instead of setting it globally and having unintentional side effects.